Microsoft.Azure.CosmosRepository/src/InMemoryRepository.cs

```diff
@@ -118,6 +121,39 @@ public async ValueTask<IEnumerable<TItem>> UpdateAsync(IEnumerable<TItem> values
             return enumerable;
         }
 
+        /// <inheritdoc/>
+        public async ValueTask UpdateAsync(string id,
+            Action<IPatchOperationBuilder<TItem>> builder,
+            string partitionKeyValue = null,
+            CancellationToken cancellationToken = default)
+        {
+            await Task.CompletedTask;
+
+            partitionKeyValue ??= id;
+
+            IItem item = Items.Values.FirstOrDefault(x => x.Id == id && x.PartitionKey == partitionKeyValue);
+
+            if (item is null)
+            {
+                NotFound();
+            }
+
+            PatchOperationBuilder<TItem> patchOperationBuilder = new();
+
+            builder(patchOperationBuilder);
+
+            foreach (InternalPatchOperation internalPatchOperation in patchOperationBuilder._rawPatchOperations)
+            {
+                if (internalPatchOperation.Type is not PatchOperationType.Replace)
+                {
+                    continue;
+                }
```

Comment:
```suggestion
            foreach (InternalPatchOperation internalPatchOperation in 
                patchOperationBuilder._rawPatchOperations.Where(ipo => ipo.Type is PatchOperationType.Replace))
            {
```

I'd use a `.Where` clause instead, less noisy. And instead of a negation, serves as a projection.